### PR TITLE
JBPM-5722: No warning when closing the editor with unsaved changes.

### PR DESCRIPTION
**Pull request by pefernan**
@wmedvede @jsoltes can you take a look?

### CODE REVIEW

File: kie-wb-common-forms/kie-wb-common-forms-core/kie-wb-common-forms-api/src/main/java/org/kie/workbench/common/forms/model/FormDefinition.java [hunk 1]
```diff
@@ -129,8 +129,11 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        int result = id != null ? id.hashCode() : 0;
-        result = ~~result;
+        int result = id.hashCode();
```
Comment (manstis) [inline kie-wb-common-forms/kie-wb-common-forms-core/kie-wb-common-forms-api/src/main/java/org/kie/workbench/common/forms/model/FormDefinition.java]: It's advisable to use ```~~``` (or other "narrowing" operation) to ensure JS's "int" does not overflow a Java "int".

There have been numerous emails to BSIG in the past about this.. happy to dig one out if you want more explanation.

File: kie-wb-common-forms/kie-wb-common-forms-core/kie-wb-common-forms-fields/src/main/java/org/kie/workbench/common/forms/fields/shared/AbstractFieldDefinition.java [hunk 2]
```diff
@@ -166,4 +166,52 @@ public void copyFrom(FieldDefinition other) {
...
+        }
+        if (binding != null ? !binding.equals(that.binding) : that.binding != null) {
+            return false;
+        }
+        return standaloneClassName != null ? standaloneClassName.equals(that.standaloneClassName) : that.standaloneClassName == null;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = id.hashCode();
```
Comment (manstis) [inline kie-wb-common-forms/kie-wb-common-forms-core/kie-wb-common-forms-fields/src/main/java/org/kie/workbench/common/forms/fields/shared/AbstractFieldDefinition.java]: Same ```hashCode()``` comment as above.

File: kie-wb-common-forms/kie-wb-common-forms-integrations/kie-wb-common-forms-jbpm-integration/kie-wb-common-forms-jbpm-integration-api/src/main/java/org/kie/workbench/common/forms/jbpm/model/authoring/JBPMVariable.java [hunk 3]
```diff
@@ -46,4 +46,28 @@ public String getType() {
...
+
+        if (!name.equals(that.name)) {
+            return false;
+        }
+        return type.equals(that.type);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = name.hashCode();
```
Comment (manstis) [inline kie-wb-common-forms/kie-wb-common-forms-integrations/kie-wb-common-forms-jbpm-integration/kie-wb-common-forms-jbpm-integration-api/src/main/java/org/kie/workbench/common/forms/jbpm/model/authoring/JBPMVariable.java]: ```hashCode()``` comment.

File: kie-wb-common-forms/kie-wb-common-forms-integrations/kie-wb-common-forms-jbpm-integration/kie-wb-common-forms-jbpm-integration-api/src/main/java/org/kie/workbench/common/forms/jbpm/model/authoring/process/BusinessProcessFormModel.java [hunk 4]
```diff
@@ -64,4 +64,28 @@ public void setProcessName(String processName) {
...
+
+        if (!processId.equals(formModel.processId)) {
+            return false;
+        }
+        return processName.equals(formModel.processName);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = processId.hashCode();
```
Comment (manstis) [inline kie-wb-common-forms/kie-wb-common-forms-integrations/kie-wb-common-forms-jbpm-integration/kie-wb-common-forms-jbpm-integration-api/src/main/java/org/kie/workbench/common/forms/jbpm/model/authoring/process/BusinessProcessFormModel.java]: And again...

File: kie-wb-common-forms/kie-wb-common-forms-integrations/kie-wb-common-forms-jbpm-integration/kie-wb-common-forms-jbpm-integration-api/src/main/java/org/kie/workbench/common/forms/jbpm/model/authoring/task/TaskFormModel.java [hunk 5]
```diff
@@ -84,4 +84,36 @@ public String getFormName() {
...
+        }
+        if (!taskName.equals(that.taskName)) {
+            return false;
+        }
+        return formName.equals(that.formName);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = processId.hashCode();
```
Comment (manstis) [inline kie-wb-common-forms/kie-wb-common-forms-integrations/kie-wb-common-forms-jbpm-integration/kie-wb-common-forms-jbpm-integration-api/src/main/java/org/kie/workbench/common/forms/jbpm/model/authoring/task/TaskFormModel.java]: And one for luck ;-)

File: kie-wb-common-forms/kie-wb-common-forms-core/kie-wb-common-forms-api/src/main/java/org/kie/workbench/common/forms/model/FormDefinition.java (hunk 1 above)
Comment (pefernan) [inline kie-wb-common-forms/kie-wb-common-forms-core/kie-wb-common-forms-api/src/main/java/org/kie/workbench/common/forms/model/FormDefinition.java]: @manstis thanks changing them...

**Comment by pefernan** [comment]
@manstis better now?

**Comment by wmedvede** [comment]
@pefernan   @manstis looks good, +1 from my side.

**Pull request merged**